### PR TITLE
Dont proxy images for bots, fixes #4448

### LINKED ISSUE #4448: Proxy Enhancement:  Don't serve up bots
After enabling the proxy on my 2.0 site, I noticed CPU up 2x & bandwidth up 3x.  I believe the crux of the problem is serving up images for search bots.  

I believe we should NOT serve up images thru the proxy for suspected bots.  

We probably shouldn't be serving up these images (originating from other sites) for search engines anyway.

This impacts 2.0 & 2.1.

### PR DESCRIPTION
**Pull request by sbulen**
Don't proxy images for bots.  

This fixes issues where site recrawls, even by good bots like Google, can cause your cache to grow by many GB of data & get completely randomized - useless as a cache.   

Also, bots probably should "see" the image's proper link anyway.  Give the image's original source page proper credit.

At this point in time, Google does not penalize sites for mixed-content pages (http & https), so this will not hurt Google's ranking of your forum's content.

### CODE REVIEW

File: Sources/Subs.php [hunk 1]
```diff
@@ -1413,7 +1413,10 @@ function parse_bbc($message, $smileys = true, $cache_id = '', $parse_tags = arra
 				'content' => '<img src="$1" alt="{alt}" title="{title}"{width}{height} class="bbc_img resized">',
 				'validate' => function (&$tag, &$data, $disabled)
 				{
-					global $image_proxy_enabled, $image_proxy_secret, $boardurl;
+					global $image_proxy_enabled, $image_proxy_secret, $boardurl, $user_info;
+
+					if (!empty($user_info['possibly_robot']))
+						return;
```
Comment (Sesquipedalian) [inline Sources/Subs.php]: This should go inside the `if ($image_proxy_enabled)` block.

File: Sources/Subs.php [hunk 2]
```diff
@@ -1436,7 +1439,10 @@ function parse_bbc($message, $smileys = true, $cache_id = '', $parse_tags = arra
 				'content' => '<img src="$1" alt="" class="bbc_img">',
 				'validate' => function (&$tag, &$data, $disabled)
 				{
-					global $image_proxy_enabled, $image_proxy_secret, $boardurl;
+					global $image_proxy_enabled, $image_proxy_secret, $boardurl, $user_info;
+
+					if (!empty($user_info['possibly_robot']))
+						return;
```
Comment (Sesquipedalian) [inline Sources/Subs.php]: This should go inside the `if ($image_proxy_enabled)` block.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

**Comment by sbulen** [comment]
I'll make the change & retest.

**Comment by sbulen** [comment]
Done.  Tests OK.
Tested on a wamp server, php5.6, MySQL 5.7.

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**